### ISSUE TITLE
Incorrect filename (uses the package name instead of the folder name)

### ISSUE THREAD

**Issue by CLOVIS-AI** [razvn/jacoco-to-cobertura-gradle-plugin#24]
Hi! I'm trying to use this plugin in a Kotlin sample project before deploying it to my real projects. It generates the file correctly, but the `filename` is incorrect.

Here is the sample code: `examples/kotlin/core/src/commonMain/kotlin/HelloWorld.kt`:
```kotlin
package example

const val HELLO_WORLD = "Hello world!"

/**
 * Empty class with documentation.
 */
class HelloWorld

fun foo() {
	println("TMP DO NOT MERGE THIS")
	println(HELLO_WORLD)
}
```

The file is directly in the `kotlin` directory (no subdirectories for the `example` package), as per the [Kotlin code conventions](https://kotlinlang.org/docs/coding-conventions.html#directory-structure).

However, the coverage report generated by this plugin looks like:
```xml
<?xml version="1.0" encoding= "UTF-8" ?>
<coverage line-rate="1.0" branch-rate="0.0" complexity="0.0" timestamp="0">
   <sources>
      <source>.</source>
   </sources>
   <packages>
      <package line-rate="1.0" branch-rate="0.0" complexity="0.0" name="example">
         <classes>
            <class line-rate="1.0" branch-rate="0.0" complexity="0.0" name="example/HelloWorld" filename="example/HelloWorld.kt">
               <methods>
                  <method line-rate="1.0" branch-rate="0.0" complexity="0.0" name="&lt;init&gt;" signature="()V">
                     <lines>
                        <line number="8" hits="1" branch="false"/>
                        <line number="11" hits="1" branch="false"/>
                     </lines>
                  </method>
               </methods>
            </class>
            <class line-rate="1.0" branch-rate="0.0" complexity="0.0" name="example/HelloWorldKt" filename="example/HelloWorld.kt">
               <methods>
                  <method line-rate="1.0" branch-rate="0.0" complexity="0.0" name="foo" signature="()V">
                     <lines>
                        <line number="8" hits="1" branch="false"/>
                        <line number="11" hits="1" branch="false"/>
                     </lines>
                  </method>
               </methods>
            </class>
         </classes>
      </package>
   </packages>
</coverage>               
```

The file name is reported as `filename="example/HelloWorld.kt"`, which is wrong.

---

Reproduction:
- `git clone https://gitlab.com/opensavvy/automation/gradle-conventions.git`
- `git switch 30-cobertura-xml-report`
- `./gradlew :jacocoToCobertura -p examples/kotlin --include-build ../..`

Files of interest:
- Plugin configuration: `root/src/main/kotlin/root.gradle.kts` ([online](https://gitlab.com/opensavvy/automation/gradle-conventions/-/merge_requests/39/diffs#daa6970f4cc24fbfb28fd69609bfb16683466f47))
- Code being tested: `examples/kotlin/core/src/commonMain/kotlin/HelloWorld.kt` ([online](https://gitlab.com/opensavvy/automation/gradle-conventions/-/merge_requests/39/diffs#4d87fd1d49931dbdc90d40caefe12392fc1786eb))
- Tests: `examples/kotlin/core/src/commonTest/kotlin/HelloWorldTest.kt` ([online](https://gitlab.com/opensavvy/automation/gradle-conventions/-/blob/f2f3d3e6f9dfbe0caaad464aa95e24679b090d9f/examples/kotlin/core/src/commonTest/kotlin/HelloWorldTest.kt))

Versions:
- Kover 0.7.6
- Jacoco to Cobertura 1.2.0

For context, I want to display coverage reports generated by Kover into the [GitLab Merge Request UI](https://docs.gitlab.com/ee/ci/testing/test_coverage_visualization.html), but [my attempt](https://gitlab.com/opensavvy/automation/gradle-conventions/-/merge_requests/39) currently doesn't work. I believe the file name being incorrect is the cause.

**Comment by razvn** [comment]
Well, is supposed to do that... at least cover2cover does that... can't know if is a kotlin or java project and if the package is `virtual` and not in a directory as should be in a java project. Nevertheless to fix that I added in `1.3.0` release (to come soon) an new option `rootPackageToRemove` that will strip the packages in the start of the filename. 

So if you set this value to `com.exemple` and your file is with package `com.example` and named `HelloWorld.kt` then. the result filename will be `HelloWord.kt`

I tried to see the result of the python script on your exemple but it does not run at all as kover Jacoco report does not seem standard.

I hope it fix your problem.

**Comment by razvn** [comment]
This project is based on the python script `cover2cover` which works that way. 

The only input used is the Jacoco report (no parsing or verification of actual project files is done) 
To guess the location of the file the package name is used, because in standard Java (and usually also Kotlin) this the packages names matches the directories structure. So the file location is guessed base on the the package. 

If you look at the Jacoco we only got the classname and the sourcefile, from that we have to guess the location of the file because Cobertura expects the full name. How to know that you really put the HelloWorld.kt in the root dir or in `./example` ? 
Change the file location to `example/HelloWorld.kt` you'll see the Jacoco/Kover report will be the same.

I didn't see Kover claiming that their report is Jacoco compliant so not sure how useful will be the issues. Didn't look much in the cover2cover error at that time, after fixing the divide by zero it seems to pass but and here's its result (so maybe Kover XML report is Jacoco compliant)

```xml
<?xml version="1.0" ?>
<coverage line-rate="1.0" branch-rate="0" complexity="0.0">
	<sources>
		<source>examples/kotlin/app/src/jvmMain/kotlin</source>
	</sources>
	<packages>
		<package name="example" line-rate="1.0" branch-rate="0" complexity="0.0">
			<classes>
				<class name="example.HelloWorld" filename="example/HelloWorld.java" line-rate="1.0" branch-rate="0" complexity="0.0">
					<methods>
						<method name="&lt;init&gt;" signature="()V" line-rate="1.0" branch-rate="0" complexity="0.0">
							<lines />
						</method>
					</methods>
					<lines />
				</class>
				<class name="example.HelloWorldKt" filename="example/HelloWorldKt.java" line-rate="1.0" branch-rate="0" complexity="0.0">
					<methods>
						<method name="foo" signature="()V" line-rate="1.0" branch-rate="0" complexity="0.0">
							<lines />
						</method>
					</methods>
					<lines />
				</class>
			</classes>
		</package>
	</packages>
</coverage>
```
So as you can see it also guesses the name of the file with the path (and lets not talk about the `.java` extension :)


Anyway, with the 1.3 version you can decide to strip part of the package you don't need so it fixes you problem didn't it ?

**Comment by razvn** [comment]
Fixed in e2b4a122af8c4a381b23ce40f43edc9dbef6554e
